pkg/kubelet/cm/dra/manager.go

```diff
@@ -88,36 +97,49 @@ func (m *ManagerImpl) prepareContainerResources(pod *v1.Pod, container *v1.Conta
 			if resource := m.resources.get(claimName, pod.Namespace); resource != nil {
 				// resource is already prepared, add pod UID to it
 				resource.addPodUID(pod.UID)
+
 				continue
 			}
 
 			// Query claim object from the API server
-			resourceClaim, err := m.kubeClient.CoreV1().ResourceClaims(pod.Namespace).Get(context.TODO(), claimName, metav1.GetOptions{})
+			resourceClaim, err := m.kubeClient.CoreV1().ResourceClaims(pod.Namespace).Get(
+				context.TODO(),
+				claimName,
+				metav1.GetOptions{})
 			if err != nil {
-				return fmt.Errorf("failed to fetch ResourceClaim %s referenced by pod %s: %+v", claimName, pod.Name, err)
+				return fmt.Errorf("failed to fetch ResourceClaim %s referenced by pod %s: %w", claimName, pod.Name, err)
```

Comment:
What's the rationale for wrapping the error? Why is that better and/or needed?

See "Whether to Wrap" in https://go.dev/blog/go1.13-errors

Reply:
The rationale was to fix this golangci warning:
```
pkg/kubelet/cm/dra/manager.go:110:106: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors (errorlint)
                                return fmt.Errorf("failed to fetch ResourceClaim %s referenced by pod %s: %+v", claimName, pod.Name, err)
```

Reply:
That check is too strict. As explained in the design document for errors, there are cases where errors should not get wrapped. This is part of the API design of a package and something that a developer needs to decide on a case-by-case basis.

Reply:
should be fixed now. please, reivew